libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationViewModel.kt

```diff
@@ -23,6 +24,8 @@ import com.mapbox.navigation.dropin.extensions.attachCreated
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 internal class DropInNavigationViewModel : ViewModel() {
 
+    val store = Store()
```

Comment:
I am a fan of centralizing state. But there are some issues have led us to decentralize it. Am raising one of the issues here.

All the `*ViewModel`s instances are "shareable" with android auto. This `Store` is only being instantiated with the ViewModel at this time, which we don't have in android auto.

The mechanism to share these viewmodels state with android auto is `MapboxNavigationObserver` through the `MapboxNavigationApp`. How would we share the `Store` state and the viewmodels with android auto with this? Why not make the `Store` state be a `MapboxNavigationObserver`?

Reply:
Sharing a state with android auto is a different problem that needs to be addressed in a separate PR.
This PR addresses state fragmentation issue.

Reply:
What's the proposed solution to share state with android auto after this change?

I had a clear idea of what to do when using instances of `MapboxNavigationObserver`, but after this change I think we'd need to change `Store` to `MapboxNavigationObserver` 🤔

Reply:
I didn't look into it yet so it's hard for me to propose the solution for this.

Few ideas that come to mind:
- make Store singleton
- move Store to a service that's accessible by both Android Auto and Navigation View
- given MapboxNavigationObserver survives activity configuration changes... wrap store in some sort of StoreHolder class that can be registered with MapboxNavigationApp